### ISSUE TITLE
Try to solve CTR problem by tf-ranking

### ISSUE THREAD

**Issue by pangpang97** [tensorflow/ranking#290]
hi  I am tring to solve CTR problem by tf-ranking.  When I ran a small demo, whose data was generated by myself, I got a mistake like this 
![image](https://user-images.githubusercontent.com/24262596/132827405-60348f15-c813-420e-a89e-0c11eba49879.png)
 
 Could you help me solve this problem?  Thanks in advance.
This is my code to generate data 
https://github.com/pangpang97/tf-ranking-demo/blob/main/genterate_data.py
This is my code of training 
https://github.com/pangpang97/tf-ranking-demo/blob/main/ranking_train.py 
This is my env:
tensorflow  2.6.0
tensorflow-ranking  0.4.2.dev
python 3.6.9

**Comment by rjagerman** [comment]
I think the issue is that your label is an `int64` and TF-Ranking is expecting a `float` in the internal `tf.greater_equal` call. We should probably change this so it can accept int64 labels, I am not sure how much effort that is, so I will follow-up on that separately.

To solve your issue for now, I think you can change your [code](https://github.com/pangpang97/tf-ranking-demo/blob/main/ranking_train.py#L64) to provide float labels instead of int64 labels:
```python
label = tf.cast(label, tf.float32)
```

**Comment by pangpang97** [comment]
> I think the issue is that your label is an `int64` and TF-Ranking is expecting a `float` in the internal `tf.greater_equal` call. We should probably change this so it can accept int64 labels, I am not sure how much effort that is, so I will follow-up on that separately.
> 
> To solve your issue for now, I think you can change your [code](https://github.com/pangpang97/tf-ranking-demo/blob/main/ranking_train.py#L64) to provide float labels instead of int64 labels:
> 
> ```python
> label = tf.cast(label, tf.float32)
> ```

Thanks,  it works. 
But I found another problem: 
![image](https://user-images.githubusercontent.com/24262596/132847538-e6821980-0e4d-4a5f-8bfe-2cd2ce514011.png)
The loss is always zero.

**Comment by rjagerman** [comment]
> Does the unuaual dimension lead to my problem( loss is zero) ?

Yes, you seem to be using softmax loss together with a list of only 1 item, which is by definition zero.

You can try using a pointwise loss such as [`sigmoid_cross_entropy_loss`](https://github.com/tensorflow/ranking/blob/6cf8f70a8533ba15abbfb5f50db17cb01fc56410/tensorflow_ranking/python/keras/losses.py#L33)/[`mean_squared_loss`](https://github.com/tensorflow/ranking/blob/6cf8f70a8533ba15abbfb5f50db17cb01fc56410/tensorflow_ranking/python/keras/losses.py#L34), or, change your data format so that it generates lists of item, which will allow you to leverage the pairwise/listwise ranking losses.

**Comment by pangpang97** [comment]
> > Does the unuaual dimension lead to my problem( loss is zero) ?
> 
> Yes, you seem to be using softmax loss together with a list of only 1 item, which is by definition zero.
> 
> You can try using a pointwise loss such as [`sigmoid_cross_entropy_loss`](https://github.com/tensorflow/ranking/blob/6cf8f70a8533ba15abbfb5f50db17cb01fc56410/tensorflow_ranking/python/keras/losses.py#L33)/[`mean_squared_loss`](https://github.com/tensorflow/ranking/blob/6cf8f70a8533ba15abbfb5f50db17cb01fc56410/tensorflow_ranking/python/keras/losses.py#L34), or, change your data format so that it generates lists of item, which will allow you to leverage the pairwise/listwise ranking losses.

Yes, I  used a  pointwise loss(acttually, SIGMOID_CROSS_ENTROPY_LOSS) , then I got a non-zero loss.  Thanks for your help.
